sdlib/d/gc/page.d

```diff
@@ -706,6 +709,7 @@ private:
 
 				import d.gc.slab;
 				auto nslots = binInfos[sc].nslots;
+				auto ssize = binInfos[sc].slotSize;
```

Comment:
If the block has a finalizer, it should have its actual size in its metadata.

Reply:
When looking up the metadata, I have an address and an index. The calculation to get the actual block pointer is:

```d
auto ptr = e.address + idx * ssize;
```

Is there a better way to do this? How do I get at the metadata before knowing the slot boundaries?

Reply:
I see. but you only need that when destroying, right? So why is this here?

Reply:
Moved until later, where there are definitely going to be finalizers called. I was thinking it would save some calculation while looping over the bits, since the slot size does not change, and I thought this was instead of doing it per finalizer. But I can do it per nimble instead, so it will be less impactful.